### PR TITLE
Docs: Rework the concept section of the Diataxis doctype overview document

### PR DESCRIPTION
**Pull request by inoxx03**

### CODE REVIEW

**Comment by inoxx03** [comment]
@ebullient Hey:) I scrapped my concept template and worked the information from int into the concept section of the doc type overview document.

I like the bullet-point format  btw:) I think it is brief and to-the-point.
I can take a crack at the other sections of the doc, too.

Please let me know what you think. Thank you:)

**Comment by ebullient** [comment]
Will bring this in atop #25354

**Comment by ebullient** [comment]
@inoxx03 -- changes should be made to the concept doc (rather than the template), specifically: https://github.com/quarkusio/quarkus/blob/main/docs/src/main/asciidoc/doc-concepts.adoc

**Comment by inoxx03** [comment]
Hi, @ebullient :) Thanks for spotting this. My bad for misplacing the info. I added the revisions to `docs/src/main/asciidoc/doc-concepts.adoc`, too.

File: docs/src/main/asciidoc/_templates/doc-concept.adoc [hunk 1]
```diff
@@ -0,0 +1,80 @@
+////
```
Comment (ebullient) [inline docs/src/main/asciidoc/_templates/doc-concept.adoc:1]: This file should be removed. The _templates folder should only contain templates to be copied.

**Review by ebullient — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/_templates/doc-concept.adoc (hunk 1 above)
Comment (inoxx03) [reply docs/src/main/asciidoc/_templates/doc-concept.adoc]: @ebullient Thanks! Agreed! ...and removed.

I also squashed the "fix-up" commits into the first one.

**Review by ebullient — APPROVED** [review]

**Pull request merged**